### DISCUSSION TITLE
Filecoin - Wallaby Testnet

### DISCUSSION THREAD

**Discussion by irwingtello** [wagmi-dev/references#10] [Ideas]
Hi , i will add a proposal to this repo, to add  the Filecoin config to connect to the wallaby testnet

**Comment by irwingtello** [comment]
I'm wondering how to add this code:

import { Chain } from 'wagmi'
 
export const filecoinWallabyTestnet: Chain = {
  id: 31415,
  name: 'Filecoin Wallaby Testnet',
  network: 'wallaby testnet',
  nativeCurrency: {
    decimals: 18,
    name: 'Filecoin',
    symbol: 'Fil',
  },
  rpcUrls: {
    default: { http: ['https://wallaby.node.glif.io/rpc/v0'] },
  },
  blockExplorers: {
    default: { name: 'Glif', url: 'https://explorer.glif.io/wallaby' },
  },
}

**Comment by jxom** [comment]
Sounds good to me. Can we also add `filecoin` & `filecoinBuildernet` (testnet) chains too?

**Reply by irwingtello** [reply]
Yes, we can add , I saw a tutorial on the Wallaby testnet, I think it is the most popular one now before the FEVM release on mainnet in February .

https://github.com/filecoin-project/testnet-wallaby


https://youtu.be/A3adihIgNfc

I'm implementing this code, because I'm building a prototype, but I'm figuring out how to add this to my code, on custom chains and custom node providers (like QuickNode) (I might propose this in another thread).

**Reply by irwingtello** [reply]
For tools on the FEVM,are grants available, i don't know how much time take to add this feature to the repo,but can be helpful

https://spacewarp.fvm.dev/

I'm working on add this feature in my project, but i would love to contribute to this repo, i'm part of community (dfhcommunity.com) dedicated to open source.

**Reply by irwingtello** [reply]
I have added the implementation on the FEVM (Wallaby) in my project,
https://github.com/irwingtello/poh/commit/103f7b17a9536c6ff3d42653209e287397a064c5

I don't know if this integration are the right way, I'm reading the documentation
